### DISCUSSION TITLE
From charge time remaining to time to complete

### DISCUSSION THREAD

**Discussion by adgoudsmit** [jheredianet/Teslamate-CustomGrafanaDashboards#36] [Ideas]
I have been searching for a way to easily make it visible at which time / date the car will be finished charging. There is the time remaining for charging, but i would like to see the time and date for the charging to be finished. Can/might this be added somewhere?

My sql query skills are not up to par for getting this myself. Suggestion would be to add this to the "Current Charge view".

**Comment by jheredianet** [comment]
Sorry, It's not possible since Teslamate does not save that remaining time for charging in the database (just export it through MQTT).

For myself I've made a program to get that info from MQTT and save it on a special table to get that value from Grafana. But as you can see, it is something that you have todo do with you setup, but it's not implemented in Teslamate by default. 

I share with you a screenshot of my "Current Charge View" to have an idea, but it's required to have not only skills on PostgreSQL but on programming on any language that could listen the MQTT protocol and write the information on an auxiliary table in TeslaMate. 

![image](https://github.com/jheredianet/Teslamate-CustomGrafanaDashboards/assets/45735191/db4e69e1-689c-4404-941c-cc7ff490bc32)

**Comment by djpalmis** [comment]
Just comment my post on official Teslamate forum: https://github.com/adriankumpf/teslamate/discussions/2750
